### PR TITLE
[IMP] stock_reserve_rule: add constraints on fallback locations

### PR DESCRIPTION
**Pull request by sebalix**

### CODE REVIEW

**Comment by sebalix** [comment]
@guewen your review on this is welcome :)

File: stock_reserve_rule/models/stock_move.py [hunk 1]
```diff
@@ -112,6 +114,19 @@ def _update_reserved_quantity(
                     owner_id=owner_id,
                     strict=strict,
                 )
+                # Then if there is still a need, we split the current move to
+                # get a new one targetting the fallback location with the
+                # remaining qties
+                still_need = self.product_uom_qty - self.reserved_availability
+                if still_need:
+                    new_move_id = self._split(still_need)
```
Comment (guewen) [inline stock_reserve_rule/models/stock_move.py]: Beware, `_split` expects the quantity as gived by `product_qty` (unit of the product) and not `product_uom_qty` (unit of the move). You have to convert it (example https://github.com/odoo/odoo/blob/939bca38212d5eba830bbf70d23f500ace578b1a/addons/stock/models/stock_move.py#L1399-L1400).

File: stock_reserve_rule/views/stock_reserve_rule_views.xml [hunk 2]
```diff
@@ -61,6 +62,10 @@
                             </group>
                         </form>
                     </field>
+                    <field name="missing_reserve_rule" invisible="1"/>
+                    <strong attrs="{'invisible': [('missing_reserve_rule', '=', False)]}" style="color:red;">
+                      WARNING: some of the removal rules are missing a reservation rule with a higher priority. If it is desirable you should create them to ensure that the reservation on the related locations will use these rules first instead of the current one.
```
Comment (guewen) [inline stock_reserve_rule/views/stock_reserve_rule_views.xml]: I don't understand what is expected :no_mouth:

File: stock_reserve_rule/models/stock_reserve_rule.py [hunk 3]
```diff
@@ -140,6 +176,44 @@ class StockReserveRuleRemoval(models.Model):
         "When empty, any packaging can be removed.",
     )
 
+    missing_reserve_rule = fields.Boolean(
```
Comment (guewen) [inline stock_reserve_rule/models/stock_reserve_rule.py:179]: Do you have details of the expectations for this behavior (some specs?), not sure to get it

File: stock_reserve_rule/views/stock_reserve_rule_views.xml (hunk 2 above)
Comment (sebalix) [reply stock_reserve_rule/views/stock_reserve_rule_views.xml]: To check with @jgrandguillaume

File: stock_reserve_rule/models/stock_reserve_rule.py (hunk 3 above)
Comment (sebalix) [reply stock_reserve_rule/models/stock_reserve_rule.py:179]: My last comment on the related card (that I just added) following a meet with Joël :)

File: stock_reserve_rule/models/stock_move.py (hunk 1 above)
Comment (sebalix) [reply stock_reserve_rule/models/stock_move.py]: indeed, fixed!

File: stock_reserve_rule/views/stock_reserve_rule_views.xml (hunk 2 above)
Comment (jgrandguillaume) [reply stock_reserve_rule/views/stock_reserve_rule_views.xml]: @guewen We spotted that without a proper information, a miss configuration could lead to a nasty behavior of the system. So for now, we inform the user about what he should configure properly.
In this case, you must define a reservation rule for each removal rule placed under stock with a lower sequence. If not, you may end up reserving goods above the location you are (e.g. reservation made in stock/input when you're in stock/higbay)

Comment (guewen) [reply stock_reserve_rule/views/stock_reserve_rule_views.xml]: I mean: as a user the message does not help me understand what I should do ;)

Comment (sebalix) [reply stock_reserve_rule/views/stock_reserve_rule_views.xml]: Should I change this message to (copy-paste from Joël comment :p)?
_You must define a reservation rule for each removal rule placed under the current location with a lower sequence. If not, you may end up reserving goods above the location you are._

**Comment by guewen** [comment]
I'll rework some parts, I'll pull your changes locally to rework them instead of merging directly.

**Pull request merged**